cmd/updatehub/main.go

```diff
@@ -132,3 +140,23 @@ func main() {
 
 	os.Exit(d.Run())
 }
+
+func sanitizeServerAddress(address string, disableHTTPS bool) (string, error) {
+	a := address
+	if !strings.HasPrefix(a, "http://") && !strings.HasPrefix(a, "https://") {
+		a = "https://" + a
+	}
+
+	serverURL, err := url.Parse(a)
+	if err != nil {
+		return "", err
+	}
+
+	if disableHTTPS {
```

Comment:
I've based the implementation on the old agent. I don't remember the reason for it. I'm changing this in a further commit.